### DISCUSSION TITLE
Any way to add a gradual tempo change?

### DISCUSSION THREAD

**Discussion by Djard** [hydrogen-music/hydrogen#1670] [Hydrogen on Windows]
I need to add a rallentando in the last measure of a song in 4/4. Has anybody found a way to slow down the tempo in a track, without the labor of writing, say, four measures in 1/4 or eight measures 1/8 time and assigning decreased tempo rate to each measure?

**Comment by theGreatWhiteShark** [comment]
Well, since the [last](https://github.com/hydrogen-music/hydrogen/discussions/1535) discussion the changes in Hydrogen's audio engine, which could enable it to support gradual tempo changes, become more and more stable. Nothing new in particular. But we are slowly moving towards that features

**Comment by Djard** [comment]
Thank you. I eagerly look forward to the development.

May I suggest another feature? Make it possible to adjust the numbers, representing bars, in the time line; so that an anacrusis (pick-up measure) is not numbered. The remainder of the incomplete measure is counted at the end of the score.

**Reply by theGreatWhiteShark** [reply]
> May I suggest another feature? Make it possible to adjust the numbers, representing bars, in the time line; so that an anacrusis (pick-up measure) is not numbered. The remainder of the incomplete measure is counted at the end of the score.

The numbers do actually count columns in the song editor in the same way it is done in e.g. Microsoft Excel. There is no direct relation with musical measure (which is a common source of confusion). In addition, Hydrogen supports MIDI and OSC commands which work with these column number. Renaming them would make things quite buggy and confusing on this end.

Have you considered using *tags* for better orientation within the song?

**Comment by trebmuh** [comment]
@Djard : it would be great to open a separate issue with another title. Doing this is helping with long time issues/discussions management.

**Reply by Djard** [reply]
Thanks.
